### PR TITLE
Heretic drip and toys

### PR DESCRIPTION
**Pull request by airlines7**
### TL;DR
Adds toy heretic sickles to donksoft vending
Adds a "fashionable robe" to the costume autodrobe as contraband, the costume version of the ominous armor.

# Description

### The drippy robe
After all these years the heretic drip now for the common man. I've made sure that it does not decrease damage as it is just a "fashionable" hood and robe. Has some silly flavor text about a massacre at a Nanotrasen fashion show.
Video proof  https://user-images.githubusercontent.com/98609667/236930600-580c9b00-b6a2-4f16-9b83-41228e526487.mp4

image

image

image

### The toy blades
I've made sure the toys do no damage and do not paralyze with the hook, I don't know why the sickles show up as random icons. I'll (probably not) figure it out and fix it at some point.

Video Proof  https://user-images.githubusercontent.com/98609667/236933452-6903479e-464e-4732-810a-f2aad86edb14.mp4

image

image

I see the "metalicr" its already fixed, too lazy to upload another photo demo.
The sickles also don't do the honk I coded in, but that just makes it more tolerable as a toy. Feel free to remove or fix.

# Changelog

:cl:  Airlines7
rscadd: Heretic costumes for the comedic clown and cynical mime.
rscadd: Toy Heretic blades, brought to you by Donksoft.

/:cl:

### CODE REVIEW

**Comment by TerraTheUnwise** [comment]
this is good, I like this.

**Comment by DylonBR** [comment]
image
theres no way this is gramatically correct, close this PR immediately and ban this individual from ever making PRs again. Yog has fallen

**Comment by airlines7** [comment]
Noticed it does conceal one's identity without an id, like the normal heretic robe. Should I keep it that way or have the robes make you always identifiable?

**Comment by ChuChu42** [comment]
amazing PR 
now i can paint my shock hand green and watch as sec think its heretic hand and cremates me

**Comment by Therandomhoboo** [comment]
Don't get why the thumbs down. As long as the grammatical errors are cleared up and the code works sure its funny to see sec shit their pants thinking its a certain gamemode.

**Comment by TerraTheUnwise** [comment]
Please merge this, too long have I had to illegally steal heretic blades from security.

**Review by MajManatee — CHANGES_REQUESTED** [review]
Fix gramatical issues, fix the icon being a cup.

**Comment by airlines7** [comment]
> Fix gramatical issues, fix the icon being a cup.

Will do.

**Comment by airlines7** [comment]
So good news, grammar is fixed. bad news, the icons being cups is a way larger issue than the scope of the pr and out of my ability
Best example of my findings: the chefdrobe, seen live in round 47479
image
I spent 4 hours messing around finding a solution and can't. 
What I did notice is the robe has an icon, I can remove the toy blades if not having icons is that big a deal.

**Comment by airlines7** [comment]
Balance council request to remove hook

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1713,3 +1714,47 @@ obj/item/toy/turn_tracker
...
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	force = 0
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_NORMAL
+	attack_verb = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "rends")
+/obj/item/gun/magic/hook/sickly_blade_toy/attack(mob/living/M, mob/living/user)
+
```
Comment (ynot01) [inline code/game/objects/items/toys.dm:1762]: Remove this empty line and put an empty line between the type definition and this proc

**Comment by airlines7** [comment]
Forgot to point out, the only changes to the map is pictured below.
image

**Review by ynot01 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1713,3 +1713,47 @@ obj/item/toy/turn_tracker
...
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	force = 0
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_NORMAL
+	attack_verb = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "rends")
+
+/obj/item/gun/magic/hook/sickly_blade/attack(mob/living/M, mob/living/user)
```
Comment (ynot01) [inline code/game/objects/items/toys.dm]: wrong path

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1713,3 +1713,47 @@ obj/item/toy/turn_tracker
...
+	lefthand_file = 'icons/mob/inhands/64x64_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/64x64_righthand.dmi'
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	force = 0
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_NORMAL
+	attack_verb = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "rends")
```
Comment (ynot01) [inline code/game/objects/items/toys.dm:1736]: the projectile is not overrided so it does 40 damage 😓

**Review by ynot01 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1734,8 +1734,9 @@ obj/item/toy/turn_tracker
 	throw_range = 5
 	w_class = WEIGHT_CLASS_NORMAL
 	attack_verb = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "rends")
+	ammo_type = null // Hook remover (tm)
```
Comment (ynot01) [inline code/game/objects/items/toys.dm]: this might runtime

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -1734,7 +1734,31 @@ obj/item/toy/turn_tracker
...
+	damage = 0
+	knockdown = 0
+	immobilize = 0 // there's no escape
+	range = 5 // hey now cowboy
+	armour_penetration = 0 // no piercing shields
+	knockdown = 0
+	hitsound = 'sound/effects/gravhit.ogg'
+
+/obj/item/projectile/hook/sickly_blade_toy/on_hit(atom/target, blocked)
+	if(ismovable(target) && blocked != 100)
```
Comment (ynot01) [inline code/game/objects/items/toys.dm:1758]: Not calling /obj/item/projectile/proc/on_hit() could have adverse effects but I'm not sure

**Comment by airlines7** [comment]
FastDMM2 defaulted to CRLF, reconverted back to LF. Also added bardrobe because that too had the icon problem.
image


> Not calling /obj/item/projectile/proc/on_hit() could have adverse effects but I'm not sure

The Original Heretic blade did this so I'm assuming nothing bad will happen. I'll test with your change and see what happens though.

**Comment by ynot01** [comment]
> > Not calling /obj/item/projectile/proc/on_hit() could have adverse effects but I'm not sure
> 
> The Original Heretic blade did this so I'm assuming nothing bad will happen. I'll test with your change and see what happens though.

The heretic blade calls the parent proc as shown here
```
/obj/item/projectile/hook/sickly_blade/on_hit(atom/target, blocked)
	. = ..() // Calls parent proc
	if(iscarbon(target) && blocked != 100)
		var/mob/living/carbon/C = target
		for(var/obj/item/shield/riot/R in C.get_all_gear())
			R.shatter() // Shield :b:roke
			qdel(R)
```

**Comment by airlines7** [comment]
> > > Not calling /obj/item/projectile/proc/on_hit() could have adverse effects but I'm not sure
> > 
> > 
> > The Original Heretic blade did this so I'm assuming nothing bad will happen. I'll test with your change and see what happens though.
> 
> The heretic blade calls the parent proc as shown here
> 
> ```
> /obj/item/projectile/hook/sickly_blade/on_hit(atom/target, blocked)
> 	. = ..() // Calls parent proc
> 	if(iscarbon(target) && blocked != 100)
> 		var/mob/living/carbon/C = target
> 		for(var/obj/item/shield/riot/R in C.get_all_gear())
> 			R.shatter() // Shield :b:roke
> 			qdel(R)
> ```

Calling for parent proc makes the hook have you bring people to the user. Thats why its not there.

**Comment by ynot01** [comment]
> Calling for parent proc makes the hook have you bring people to the user. Thats why its not there.

Correct

**Comment by airlines7** [comment]
After fiddling with the sickles I can assure that not calling the proc for the hook does nothing. If for some reason it does it can be altered in the future. If something does happen, know that the attack log keep who shot the hook still.

**Comment by airlines7** [comment]
I will be splitting the PR into two, one for the robes, and one for the toys, the toys are obviously taking a lot longer to deal with and the robes are in a working state.
